File: hackerrank/problem solving/algorithms/1 - warmup/5 - a very big sum/a_very_big_sum.py
# https://www.hackerrank.com/challenges/a-very-big-sum/problem?isFullScreen=true

def main():
    input()
    arr = read_numbers()
    print(a_very_big_sum(arr))


def read_numbers():
    return list(map(int, input().split()))


# n: length of arr
# T: O(n)
# S: O(1) extra space
def a_very_big_sum(arr):
    return sum(arr)


if __name__ == '__main__':
    main()

File: hackerrank/problem solving/algorithms/1 - warmup/5 - a very big sum/a_very_big_sum.js
// https://www.hackerrank.com/challenges/a-very-big-sum/problem?isFullScreen=true

function main() {
  readLine();
  const arr = readNums();
  console.log(aVeryBigSum(arr));
}

function readNums() {
  return readLine().split(' ').map(Number);
}

// n: length of arr
// T: O(n)
// S: O(1) extra space
function aVeryBigSum(arr) {
  return arr.reduce((a, b) => a + b);
}

////////////////////////////////////////////////

const readline = require('readline');

const rl = readline.createInterface({ input: process.stdin });
const inputLines = [];
rl.on('line', (line) => inputLines.push(line));
rl.on('close', main);

function readLine() {
  return inputLines.shift();
}
